### ISSUE TITLE
Not able to create Issue Alert Rule to use dynamic slack channel_name

### ISSUE THREAD

**Issue by onemindatime** [getsentry/sentry#24795]
## Important Details

How are you running Sentry?

Saas (sentry.io)

## Description

Hello there, We just started using Sentry at our company and are evaluating if it is a viable option for all of our products. We have many teams that owns different applications. We would like to setup the Sentry Alerts in a way that each team can control which slack channel they want to get the notifications when there are errors in the corresponding product areas. While integrating Slack with Sentry, I tried to set a custom tag "channel_name" in our code when Sentry events are sent, and use that tag to send alerts to that particular slack channel dynamically. 

Create Alerts UI doesn't seem to have an option to do this. Based on this github issue it seems like this feature was added couple of years ago: https://github.com/getsentry/sentry-plugins/issues/138. Is this possible with the latest version of Sentry?

I did send a support ticket a week ago and haven't received a response yet.

Thanks!

## Steps to Reproduce

1. Create a New Alert Rule
2. Select Issue Alert
3. Pick any Alert condition
4. In the "Then perform these actions" step, pick "Send a Slack notification"
5. Select a Slack workspace and try to enter a slack channel_name that doesn't exist in the workspace
6. Click "Create Alert Rule" or "Save"

For e.g., 
![image](https://user-images.githubusercontent.com/40774045/112884488-69cc1f80-9084-11eb-845a-5328b28bce27.png)

### What you expected to happen

We should be able to set the channel_name based on the tags or other context that is passed through the sentry events. The tags are already accessible in the same UI in the filter condition section.
![image](https://user-images.githubusercontent.com/40774045/112885077-23c38b80-9085-11eb-9e85-5926540c644f.png)

I guess there is a possibility that the channel_name may not exist in which case there won't be any alert sent. Sentry's alerts can be treated like a Fire and Forget kind of mechanism. Because the same is true if the channel is renamed/deleted after the "Alert Rule" is created.

**Comment by BYK** [comment]
Ping @getsentry/ecosystem for triage. Also ping @getsentry/workflow in case this falls under their umbrella.

**Comment by onemindatime** [comment]
Thanks @BYK Could you pls tell me how/where do I ping the sentry teams? is it just in github, slack,..?

**Comment by wedamija** [comment]
@onemindatime Sorry, at the moment we don't support dynamic channel names based on tags with our slack integration. What you can do instead is set up a rule for each team that checks for the specific tags in the filter, and then has the team's slack channel hardcoded in the action.

I don't think we have any plans to add this in the near future - I'll leave it to @getsentry/ecosystem to comment on that part.

**Comment by MeredithAnya** [comment]
> I don't think we have any plans to add this in the near future 

Yeah I don't see this being on our roadmap anytime soon, and not sure we'd be able to do this since I don't see a great way for us to be able to verify permissions for the channels without saving them first.  Maybe @jonesphillip or @manuzope can speak to this more.

**Comment by BYK** [comment]
@onemindatime 
>Thanks @BYK Could you pls tell me how/where do I ping the sentry teams? is it just in github, slack,..?

Sorry for the confusion, my message was to ping them (the @ mentions pings the teams) 🙂

**Comment by onemindatime** [comment]
Thanks @wedamija  @MeredithAnya @BYK 
@wedamija That's pretty much what I am doing right now. Unfortunately it doesn't scale well if we have to constantly add new packages/teams to our org.

@MeredithAnya I also agree with issues around checking for permissions. But I was wondering if that could be a user option to agree that the channel existence/permissions will be evaluated at runtime (or not at all), and/or they may not get any alerts. I agree that it's not obvious why someone is not getting any alerts. In this case, it will be a user opt-in configuration.

The only reason why I'm saying that is because after I create a rule (with the channels permission evaluated), if I go into slack and delete the channel, nothing will happen to the Alert Rule (no errors anywhere in Sentry UI) until I edit the rule and try to update it again.

Really appreciate the responses. Thanks again!

**Comment by jonesphillip** [comment]
@onemindatime this is an interesting use case. We are planning to work on letting users define team based notification channels. That way when an issue is [owned](https://docs.sentry.io/product/issues/issue-owners/) or assigned to a certain team/person we'll be able to route the alerts accordingly. Does that sound like something you'd be interested in? If so we can keep you up to date.

**Comment by onemindatime** [comment]
Thanks @jonesphillip. Yes, that sounds like a great feature. It would help redirect the notifications to appropriate channels provided we create those rules first. But our main problem is Scale. If we are adding more packages/modules over time, we will have to continue add to new Ownership rules and keep tweaking it when we need to change channels for example. We would like to just setup a base rule once and let the teams determine where they want the notifications to be sent. That way we don't have to maintain configurations in multiple systems. 

I understand if this feature request is not a high priority or not in your roadmap. Appreciate the responses!

**Comment by snoozbuster** [comment]
+1 on this. When a frontend engineer opens a PR, our build system automatically deploys that branch to a custom URL for preview and review, and we are tagging all sentry events logged from that branch with both the branch name and the code owner's slack handle. I'd like to configure a Sentry alert rule to automatically DM the code owner using this tag - but since dynamic channel names aren't supported (DMs are, however!), there's no way I can do this without setting up one rule for each engineer.

**Comment by ceorourke** [comment]
Hi @onemindatime we did add [team notifications](https://docs.sentry.io/product/integrations/notification-incidents/slack/#team-notifications) for Slack, I know it's not exactly what you were after but does this meet your needs? You can select which team to notify and if the team has a Slack channel set up it'll be routed to it.

**Comment by chfleury** [comment]
+1. It would be perfect if I could have a single Alert rule where, when the tag with the key "slack_redirect" is defined, the message would be sent to the channel with the value of the tag. Instead, I need multiple Alert rules, each mapping a tag value and sending the message to the respective channel.

The team notifications feature would also work for my use case, but it needs more setup.

**Comment by gaprl** [comment]
Hey @onemindatime, this issue has been open for over 3 years. Is this still a feature you need? We've had several community members express interest in this capability. If this is still relevant, could you confirm your current use case?

**Comment by onemindatime** [comment]
Hey Gabriel, yes, this is still very much an open issue on our side that we are not able to use dynamic slack channel names. Right now we have to give admin previleges to many users sometimes just for this purpose where they can maintain their team alerts.

Thanks

**Comment by kpujjigit** [comment]
Adding a related/unresolved/more-general angle to this on behalf of an enterprise client:

Beyond routing by a dynamic tag/channel_name, there's a broader "runtime Slack alerting config" gap: Sentry alert rules currently bind to a Slack channel ID resolved at rule-save time, with no ongoing validation. In orgs where teams and their Slack channels change frequently (reorgs, quarterly team restructuring, channel archival/replacement), this creates two compounding problems:

1. Setup is invisible from code. A team can wire up their service to report to Sentry entirely in code/config, but alerts won't actually reach Slack until someone also manually creates an Issue/Metric Alert Rule in the Sentry UI and adds a "Send a Slack notification" action. There's no signal in-repo or in CI that a corresponding Issue/Metric Alert is missing; therefore it's easy for a team to believe alerting is live when it silently isn't.
2. Config rot is silent. Because the rule stores a resolved channel ID (not a live name lookup), a channel rename is harmless, but if the channel is archived, deleted, or replaced (common when teams reorganize), the alert rule keeps referencing a dead ID with no warning back to the owning team. The "channel ID is stable" property gives a false sense of durability: it survives renames but not the far more common failure mode of channel lifecycle changes.

I see 2 approaches based on what I've dug into:
a. The original ask here (dynamic/tag-driven channel routing), or 
b. Periodic validation of existing Slack alert action targets (channel still exists/bot still has access) with a surfaced warning/notification to the rule owner when it doesn't. 

Either would meaningfully reduce the "set it up once, silently rots" failure mode that orgs with fluid team/channel structures hit constantly.
